runtime/sunvmi/sunvmi.c

```diff
@@ -614,12 +614,18 @@ getClassContextIterator(J9VMThread * currentThread, J9StackWalkState * walkState
 JNIEXPORT void JNICALL
 JVM_GC_Impl(void)
 {
-	J9InternalVMFunctions *vmFuncs = VM.javaVM->internalVMFunctions;
-	J9VMThread *currentThread = vmFuncs->currentVMThread(VM.javaVM);
-	J9MemoryManagerFunctions *mmFuncs = VM.javaVM->memoryManagerFunctions;
+	J9JavaVM *vm = VM.javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9VMThread *currentThread = vmFuncs->currentVMThread(vm);
+	J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
+	I_64 startTicks = 0;
+	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	Trc_SunVMI_GC_Entry(currentThread);
+	if (J9_EVENT_IS_HOOKED(vm->hookInterface, J9HOOK_SYSTEM_GC_CALLED)) {
```

Comment:
check should be done after the trace point

Reply:
actually should probably get the starttime just before the GC request is made, after getting VMaccess